Optional dependencies tests fail for aesara>=2.7, work with aesara<=2.6.6

This makes every PR fail.

@brandonwillard do you know what's going on here?

This SymPy test fails with Aesara 2.7.1 but passed with previous versions of Aesara:
```console
$ pytest sympy/printing/tests/test_aesaracode.py::test_MatrixSlice --pdb
============================================================= test session starts =============================================================
platform darwin -- Python 3.10.0, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
architecture: 64-bit
cache:        yes
ground types: python 

rootdir: /Users/enojb/current/sympy/sympy, configfile: pytest.ini
plugins: cov-3.0.0
collected 1 item                                                                                                                              

sympy/printing/tests/test_aesaracode.py F
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> traceback >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

    def test_MatrixSlice():
        from aesara.graph.basic import Constant
    
        cache = {}
    
        n = sy.Symbol('n', integer=True)
        X = sy.MatrixSymbol('X', n, n)
    
        Y = X[1:2:3, 4:5:6]
        Yt = aesara_code_(Y, cache=cache)
    
        s = Scalar('int64')
        assert tuple(Yt.owner.op.idx_list) == (slice(s, s, s), slice(s, s, s))
        assert Yt.owner.inputs[0] == aesara_code_(X, cache=cache)
        # == doesn't work in Aesara like it does in SymPy. You have to use
        # equals.
>       assert all(Yt.owner.inputs[i].equals(Constant(s, i)) for i in range(1, 7))
E       assert False
E        +  where False = all(<generator object test_MatrixSlice.<locals>.<genexpr> at 0x10f7d6c70>)

sympy/printing/tests/test_aesaracode.py:425: AssertionError
```
From the debugger I can see that instead of `Constant(s, i)` we have `ScalarConstant(s, i)`:
```python
-> assert all(Yt.owner.inputs[i].equals(Constant(s, i)) for i in range(1, 7))
(Pdb) l
420  	    s = Scalar('int64')
421  	    assert tuple(Yt.owner.op.idx_list) == (slice(s, s, s), slice(s, s, s))
422  	    assert Yt.owner.inputs[0] == aesara_code_(X, cache=cache)
423  	    # == doesn't work in Aesara like it does in SymPy. You have to use
424  	    # equals.
425  ->	    assert all(Yt.owner.inputs[i].equals(Constant(s, i)) for i in range(1, 7))
426  	
427  	    k = sy.Symbol('k')
428  	    aesara_code_(k, dtypes={k: 'int32'})
429  	    start, stop, step = 4, k, 2
430  	    Y = X[start:stop:step]
(Pdb) p Yt.owner
Subtensor{int64:int64:int64, int64:int64:int64}(X, ScalarConstant{1}, ScalarConstant{2}, ScalarConstant{3}, ScalarConstant{4}, ScalarConstant{5}, ScalarConstant{6})
(Pdb) p Yt.owner.inputs[1]
ScalarConstant{1}
(Pdb) p Constant(s, 1)
Constant{1}
(Pdb) p Yt.owner.inputs[1].equals(Constant(s, 1))
False
(Pdb) p type(Yt.owner.inputs[1])
<class 'aesara.scalar.basic.ScalarConstant'>
(Pdb) p type(Constant(s, 1))
<class 'aesara.graph.basic.Constant'>
```
Is that something that was intentionally changed in Aesara?

--- a/sympy/printing/tests/test_aesaracode.py
+++ b/sympy/printing/tests/test_aesaracode.py
@@ -407,8 +407,6 @@ def test_slice():
     assert theq_slice(aesara_code_(slice(1, x, 3), dtypes=dtypes), slice(1, xt, 3))
 
 def test_MatrixSlice():
-    from aesara.graph.basic import Constant
-
     cache = {}
 
     n = sy.Symbol('n', integer=True)
@@ -422,7 +420,7 @@ def test_MatrixSlice():
     assert Yt.owner.inputs[0] == aesara_code_(X, cache=cache)
     # == doesn't work in Aesara like it does in SymPy. You have to use
     # equals.
-    assert all(Yt.owner.inputs[i].equals(Constant(s, i)) for i in range(1, 7))
+    assert all(Yt.owner.inputs[i].data == i for i in range(1, 7))
 
     k = sy.Symbol('k')
     aesara_code_(k, dtypes={k: 'int32'})

